front/src/modules/scenario/components/ScenarioExplorer/ScenarioExplorer.tsx

```diff
@@ -73,12 +92,22 @@ const ScenarioExplorer = ({
     return validTrains.length;
   };
 
+  const v2TrainCount = (trainIds: number[]) => trainIds.length;
+
+  const scenario = useMemo(
+    () => (trainScheduleV2Activated ? scenarioV2 : scenarioV1),
+    [trainScheduleV2Activated, scenarioV2, scenarioV1]
+  );
+
   useEffect(() => {
-    if (scenarioDetails?.timetable_id) {
-      dispatch(updateTimetableID(scenarioDetails.timetable_id));
-      dispatch(updateInfraID(scenarioDetails.infra_id));
+    if (scenarioV1?.timetable_id) {
+      dispatch(updateTimetableID(scenarioV1.timetable_id));
+      dispatch(updateInfraID(scenarioV1.infra_id));
+    } else if (scenarioV2?.timetable_id) {
+      dispatch(updateTimetableID(scenarioV2.timetable_id));
+      dispatch(updateInfraID(scenarioV2.infra_id));
```

Comment:
You can directly check scenario:
```suggestion
    if (scenario) {
      dispatch(updateTimetableID(scenario.timetable_id));
      dispatch(updateInfraID(scenario.infra_id));
    }
  }, [scenario]);
```